Examples/Scale_Microgrid/Scale_Microgrid.cpp

```diff
@@ -0,0 +1,317 @@
+        model_id++, RN);
+
+        virDQbus_model->setExternalConnectionNodes(0, vdqbus_index[i]);
+        virDQbus_model->setExternalConnectionNodes(1, vdqbus_index[i] + 1);
+        sysmodel->addComponent(virDQbus_model);
+    }
+    
+    //allocate all the intial conditions
+    sysmodel->allocate(vec_size_total);
+
+    std::cout << sysmodel->y().size() << std::endl;
+    std::cout << vec_size_internals << ", " << vec_size_externals << "\n";
+
+    //Create Intial points for states. Every state is to specified to the zero intially
+    for (size_t i = 0; i < vec_size_total; i++)
+    {
+        sysmodel->y()[i] = 0.0;
+        sysmodel->yp()[i] = 0.0;
+    }
+
+    // Create Intial derivatives specifics generated in MATLAB
+    for (size_t i = 0; i < 2*Nsize; i++)
+    {
+        sysmodel->yp()[13*i - 1 + 3] = DGParams_list[i].Vn_;
+        sysmodel->yp()[13*i - 1 + 5] = DGParams_list[i].Kpv_ * DGParams_list[i].Vn_;
+        sysmodel->yp()[13*i - 1 + 7] = (DGParams_list[i].Kpc_ * DGParams_list[i].Kpv_ * DGParams_list[i].Vn_) / DGParams_list[i].Lf_;
+    }
+
+    //since the intial P_com = 0, the set the intial vector to the reference frame
+    sysmodel->y()[vec_size_internals] = DG_parms1.wb_;
+
+    sysmodel->initialize();
+    sysmodel->evaluateResidual();
+
+    std::vector<double>& fres = sysmodel->getResidual(); 
+    std::cout << "Verify Intial Resisdual is Zero: {\n";
+    for (size_t i = 0; i < fres.size(); i++)
+    {
+        printf("%lu : %.16e \n", i, fres[i]);
+    }
+    std::cout << "}\n";
+
+    sysmodel->updateTime(0.0, 1.0e-8);
+    sysmodel->evaluateJacobian();
+    std::cout << "Intial Jacobian with alpha:\n";
+
+
+    //Create numerical integrator and configure it for the generator model
+    AnalysisManager::Sundials::Ida<double, size_t>* idas = new AnalysisManager::Sundials::Ida<double, size_t>(sysmodel);
+
+    double t_init  = 0.0;
+    double t_final = 1.0;
+
+    // setup simulation
+    idas->configureSimulation();
+    idas->getDefaultInitialCondition();
+    idas->initializeSimulation(t_init);
+
+    idas->runSimulation(t_final);
+
+    std::vector<double>& yfinial = sysmodel->y(); 
+
+    std::cout << "Final Vector y\n";
+    for (size_t i = 0; i < yfinial.size(); i++)
+    {
+        printf("%lu: % 2.16e\n", i, yfinial[i]);
+    }
+
+    //if proper MATLAB results are avalible and model is using origional parameters
+    std::vector<size_t> compDataSizes = {2,4,8};
+    //All data generated with abstol=1e-12 and reltol=1e-12 with ode23tb
+    if (orgparams && compDataSizes.end() != std::find(compDataSizes.begin(), compDataSizes.end(), Nsize))
+    {   
+        //get the cmake source dir from the cmake configuration
+        std::string base_path = XSTRING(SOURCE_ROOT);
+
+        base_path += "/Examples/Scale_Microgrid/microgrid_N";
+        base_path += std::to_string(Nsize);
+        base_path += ".txt";
+        std::cout << base_path << std::endl;
```

Comment:
Instead of this, use current directory as a default and allow user to specify directory as an input parameter. Also, see `ParameterEstimation.cpp` example.